server/src-lib/Hasura/Backends/Postgres/DDL/RunSQL.hs

```diff
@@ -300,7 +300,7 @@ runTxWithMetadataCheck source sourceConfig txAccess tableCache functionCache cas
         -- Before running the @'tx', fetch metadata of existing tables and functions from Postgres.
         let tableNames = M.keys tableCache
             computedFieldFunctions = concatMap getComputedFieldFunctions (M.elems tableCache)
-            functionNames = nub $ M.keys functionCache <> computedFieldFunctions
+            functionNames = OHS.fromList $ M.keys functionCache <> computedFieldFunctions
```

Comment:
If we're using `HashSet` instead, this could be:

```suggestion
            functionNames = M.keysSet functionCache <> HS.fromList computedFieldFunctions
```